service/rtc/turn.go

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) 2022-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package rtc
+
+import (
+	"crypto/hmac"
+	"crypto/sha1"
+	"encoding/base64"
+	"fmt"
+	"time"
+)
+
+func genTURNCredentials(username, secret string, expirationTS int64) (string, string, error) {
+	if username == "" {
+		return "", "", fmt.Errorf("username should not be empty")
+	}
+
+	if secret == "" {
+		return "", "", fmt.Errorf("secret should not be empty")
+	}
+
+	if expirationTS <= 0 {
+		return "", "", fmt.Errorf("expirationTS should be a positive number")
```

Comment:
should a maximum age be imposed?

Reply:
It won't hurt I guess but finding a value may be a bit of a guesswork. Given the recommended value is 1 day we could put a week (7 days) maybe? I can't think of any use-case to go higher than that to be honest.